pkg/discoverfiles/discoverfilest_test.go

```diff
@@ -0,0 +1,77 @@
+package discoverfiles_test
+
+import (
+	"io/fs"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/OpenSLO/oslo/pkg/discoverfiles"
+)
+
+// TestDiscoverFilePaths tests it on real filesystem,
+// content of those file doesn't matter, thus they're empty.
+func TestDiscoverFilePaths(t *testing.T) {
+	t.Parallel()
```

Comment:
Right now this `t.Parallel()` doesn't do too much. It will just wait until until all synchronous tests finish in `discoverfiles_test` and run tests marked as `t.Parallel()`(only this one) from this package. 

To really parallelize this tests I'd suggest to run all testCases as independent sub tests with `t.Parallel()` for each of them
``` go 
	for _, tC := range testCases {
	     t.Run(tC.name, func(t *testing.T){
	                t.Parallel()
	     		res, err := discoverfiles.DiscoverFilePaths(tC.filePaths, tC.recursive)
		        assert.Equal(t, tC.want, res)
		       assert.ErrorIs(t, err, tC.expectedError)
	     }
	}
```
This won't work yet because all cases will wait and will be executed for last test case. To fix that issue you need to create copy of tC for local scope. 
``` go 
	for _, tC := range testCases {
	     tC:=tC
	     t.Run(tC.name, func(t *testing.T){
	     	         t.Parallel()
	     		res, err := discoverfiles.DiscoverFilePaths(tC.filePaths, tC.recursive)
		        assert.Equal(t, tC.want, res)
		       assert.ErrorIs(t, err, tC.expectedError)
	     }
	}
```
This will create truly parallelized tests  and we saved about 0.01s:D 

P.S: Please fmt this examples for yourself. Writing them in here isn't best idea :D

Reply:
Yeah, I'm just running here the entire test (with all the cases run one by one) in parallel with other tests, I can change it to run each case in parallel too